File: src/ProjectR/content/scripts/spells/DistortionField.lua
function GetTargetType()
    return Enemies
end

function GetName()
    return "Distortion Field"
end

function GetDescription()
    return "Creates a field of arcane energy, damaging all enemies.\nSlightly ignores enemies MR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .56
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  local damage = (3.75 * aMD * (aARC/100) - 0.75 * dMR) * (100/dARC)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/DistortionField.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class DistortionField : SpellScriptBase
    {
        public override string Name { get { return "Distortion Field"; } }
        public override string Description { get { return "Creates a field of arcane energy, damaging all enemies.\nSlightly ignores enemies MR."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC,  }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 56; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.75 * cMD * (cARC/100) - .75 * tMR) * (100/tARC);
            DealDamage(damage);
        }
    }
}